daisy/step_create_instances.go

```diff
@@ -44,17 +45,22 @@ Loop:
 		case <-tick:
 			resp, err := w.ComputeClient.GetSerialPortOutput(path.Base(i.Project), path.Base(i.Zone), i.Name, port, start)
 			if err != nil {
-				// Instance is stopped or stopping.
+				numErr++
 				status, sErr := w.ComputeClient.InstanceStatus(path.Base(i.Project), path.Base(i.Zone), i.Name)
 				switch status {
 				case "TERMINATED", "STOPPED", "STOPPING":
+					// Instance is stopped or stopping.
 					if sErr == nil {
 						break Loop
 					}
 				}
-				w.LogStepInfo(s.name, "CreateInstances", "Instance %q: error getting serial port: %v", i.Name, err)
-				break Loop
+				if numErr > 10 {
```

Comment:
Is ignoring the error for 30 seconds (since this is called with a 3 second interval) too long? If serial console logs can't be retrieved due to a transient error, this might be too long to wait before showing an error.

Reply:
Parameter tuning is important for good user experience.

That said, in this case, this code is purely run in the background. So the large number of retries is to increase the probability that we'll gather all serial logs, in spite of any transient errors.

(Compare this with the waitForInstanceSignal, which *is* monitored by the main thread.)